app/src/modules/content/routes/item.vue

```diff
@@ -352,20 +352,30 @@ async function saveVersionAction(action: 'main' | 'stay' | 'quit') {
 				refresh();
 				break;
 			case 'quit':
-				if (!props.singleton) router.push(`/content/${props.collection}`);
+				redirectAfterSave();
 				break;
 		}
 	} catch {
 		// Save version shows unexpected error dialog
 	}
 }
 
+function redirectAfterSave() {
+	const returnToPreviousPage = router.currentRoute.value.query?.return;
```

Comment:
Can the query parameter be a little more explanatory matching its behavior? Like `return_on_save` or something else? Just `return` is a bit ambiguous